cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -166,10 +208,103 @@ func (s *Supervisor) loadConfig(configFile string) error {
 	return nil
 }
 
-// TODO: Implement bootstrapping https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21071
-// nolint: unparam
 func (s *Supervisor) getBootstrapInfo() (err error) {
-	s.agentVersion = "1.0.0"
+	port, err := s.findRandomPort()
+	if err != nil {
+		return err
+	}
+
+	supervisorPort, err := s.findRandomPort()
+	if err != nil {
+		return err
+	}
+
+	var cfg bytes.Buffer
+
+	err = s.bootstrapTemplate.Execute(&cfg, map[string]any{
+		"EndpointPort":   port,
+		"InstanceUid":    s.instanceID.String(),
+		"SupervisorPort": supervisorPort,
+	})
+	if err != nil {
+		return err
+	}
+
+	s.writeEffectiveConfigToFile(cfg.String(), s.effectiveConfigFilePath)
+
+	srv := server.New(s.logger.Sugar())
+
+	done := make(chan struct{}, 1)
+	var connected atomic.Bool
+
+	err = srv.Start(newServerSettings(flattenedSettings{
+		endpoint: fmt.Sprintf("localhost:%d", supervisorPort),
+		onConnectingFunc: func(request *http.Request) {
+			connected.Store(true)
+
+		},
+		onMessageFunc: func(_ serverTypes.Connection, message *protobufs.AgentToServer) {
+			if message.AgentDescription != nil {
+				s.agentDescription = message.AgentDescription
+				identAttr := s.agentDescription.IdentifyingAttributes
+
+				for _, attr := range identAttr {
+					switch attr.Key {
+					case semconv.AttributeServiceInstanceID:
+						// TODO Consider this a critical error
+						// https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/29864
+						if attr.Value.GetStringValue() != s.instanceID.String() {
+							s.logger.Error(
+								"Client's instance ID does not match with the instance ID set by the Supervisor",
+								zap.String("expected", s.instanceID.String()),
+								zap.String("saw", attr.Value.GetStringValue()),
+							)
+						}
+					case semconv.AttributeServiceName:
+						s.agentName = attr.Value.GetStringValue()
+					case semconv.AttributeServiceVersion:
+						s.agentVersion = attr.Value.GetStringValue()
+					}
+				}
+
+				done <- struct{}{}
```

Comment:
We don't validate that s.agentName and s.agentVersion are set - it's possible these are not present in `identAttr`, does this need to be solved for?

Reply:
That's a good callout. [According to the OpAMP spec](https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#agentdescriptionidentifying_attributes), we don't require that these are set. That said, I'll make sure to mention this in a comment.

Reply:
I was thinking through this and realized that while we don't need to check that these are in the AgentDescription message, we don't want to override them later on when composing the effective config. I eliminated these two fields and just compose the config for the resource attributes from the AgentDescription message directly. Let me know what you think.